pkg/file/builder.go

```diff
@@ -1033,6 +1037,65 @@ func (b *stateBuilder) rbacRoles() {
 	}
 }
 
+var (
+	IPv6HasPortPattern    = regexp.MustCompile(`\]\:\d+$`)
+	IPv6HasBracketPattern = regexp.MustCompile(`\[\S+\]$`)
+)
+
+func isIPv6(hostname string) bool {
+	parts := strings.Split(hostname, ":")
+	return len(parts) > 2
```

Comment:
This can be prone to errors or to be misused. Why not use https://pkg.go.dev/net/netip#ParseAddr and https://pkg.go.dev/net/netip#Addr.Is6 ?

Reply:
As silly as it may look, this is enough to catch all combinations of targets in IPv6 format, including the one with port and brackets like `[2600:1f16:1784:fa03:fd73::e]:1326`, which the library wouldn't consider as v6

Reply:
Wouldn't https://pkg.go.dev/net/netip#ParseAddrPort work for this then?

I'm just worried that this func is very easy to abuse.

`isIPv6("this:is:nuts:!")` will also return `true`. It will also return true for `1:2:3:4` which http://sqa.fyicenter.com/1000334_IPv6_Address_Validator.html#Result reports as invalid.

Reply:
`ParseAddrPort` works if the port is always there, but it would fail to parse things like `::1` (would work with `[::1]:8000` though).

If you really feel strong about this we can improve this validation, but generally decK (or this library) doesn't do much input validation other than what's part of the JSON schemas from go-kong. 

For the records though, those bogus entries would still fail validation at a later stage:

```
$ ./deck gateway sync kong.yaml
Error: building state: invalid ipv6 address 1:2:3:4

$ ./deck gateway sync kong.yaml
Error: building state: invalid ipv6 address this:is:nuts:!
```

Reply:
Understood that this is validated on other levels.

I'm just suggesting to consider improving the reliability of this while we're at it

For instance something like this should work just fine and will work for addresses with and without ports

```
func IsIPv6(str string) bool {
	addr, err := netip.ParseAddrPort(str)
	if err != nil {
		addr, err := netip.ParseAddr(str)
		fmt.Printf("error parsing address: %v\n", err)
		if err != nil {
			fmt.Printf("error parsing address port: %v\n", err)
			return false
		}
		return addr.Is6()
	}

	return addr.Addr().Is6()
}
```

Reply:
This still would fail to parse something like `[::1]`, as well as detecting bogus patterns like `[invalid:ipv6::address]:1326`: the function would rightly return `false`, but this is not just `false`, it's invalid. At this point we would need add extra handling and validation for these cases too. I think this is going to become more complex and fragile overall.

I'm renaming this function to `hasIPv6Format` which better represent its purpose: https://github.com/Kong/go-database-reconciler/pull/109/commits/6764fd2987b96ab1ef282abdd31c070dc3fa8b83